appium/sample-scripts/python/testdroid_android_hybrid.py

```diff
@@ -131,14 +132,12 @@ def testSample(self):
         element.click()
         self.utils.screenshot('webpage_loaded')
 
-        contexts = "undefined"
+        context = "undefined"
         end_time = time.time() + 30
-        while "undefined" in str(contexts) and time.time() < end_time:
-            contexts = self.driver.contexts
-            self.utils.log(str(contexts))
-            sleep(5)
+        while "undefined" in context and time.time() < end_time:
+            context = str(self.driver.contexts[-1])
+            self.utils.log("Available contexts: {}, picking: {}".format(self.driver.contexts, context))
```

Comment:
I would call the `self.driver.contexts` command only once to make sure the log has exactly the same information as what we got in reality in the step before.

I think I would keep this solution, but write the while statement like this:
```
        while "undefined" in context and time.time() < end_time:
            contexts = self.driver.contexts
            context = str(contexts[-1])
            self.utils.log("Available contexts: {}, picking: {}".format(contexts, context))
```